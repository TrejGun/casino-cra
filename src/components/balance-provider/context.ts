import {createContext} from "react";

export interface IBalanceContext {
  balance: number;
  update: (amount: number) => void;
}

// eslint-disable-next-line
export const BalanceContext = createContext<IBalanceContext>(undefined!);
