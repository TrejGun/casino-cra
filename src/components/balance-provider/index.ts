export type {IBalanceContext} from "./context";
export {BalanceContext} from "./context";
export {BalanceProvider} from "./provider";
