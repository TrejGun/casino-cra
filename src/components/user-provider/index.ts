export type {IUser, IUserContext} from "./context";
export {UserContext} from "./context";
export {UserProvider} from "./provider";
