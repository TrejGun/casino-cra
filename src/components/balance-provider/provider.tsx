import React, {useEffect, useState, PropsWithChildren, ReactElement} from "react";

import {BalanceContext} from "./context";

const STORAGE_NAME = "casino_balance";

interface IBalanceProviderProps {
  balance?: number;
}

export const BalanceProvider = (props: PropsWithChildren<IBalanceProviderProps>): ReactElement | null => {
  const {balance: defaultBalance = 0, children} = props;
  const [balance, setBalance] = useState<number>(defaultBalance);

  useEffect(() => {
    const auth = window.localStorage.getItem(STORAGE_NAME);
    setBalance(auth ? JSON.parse(auth) : defaultBalance);
  }, [defaultBalance]);

  const save = (key: string, balance: number): void => {
    const json = JSON.stringify(balance);
    window.localStorage.setItem(key, json);
  };

  const update = (amount: number): void => {
    setBalance(balance + amount);
    save(STORAGE_NAME, balance);
  };

  return (
    <BalanceContext.Provider
      value={{
        balance,
        update,
      }}
    >
      {children}
    </BalanceContext.Provider>
  );
};
