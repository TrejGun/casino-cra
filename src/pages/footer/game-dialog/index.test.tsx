import React from "react";
import {cleanup, render} from "@testing-library/react";

import {BalanceProvider} from "../../../components/balance-provider";
import {GameDialog} from "./";

afterEach(cleanup);

describe("<GameDialog />", () => {
  it("renders dialog", () => {
    const container = document.createElement("div");
    document.body.append(container);

    const props = {
      container,
      open: true,
      onClose: jest.fn(),
      onResults: jest.fn(),
    };

    const {asFragment} = render(
      <BalanceProvider balance={100}>
        <GameDialog {...props} />
      </BalanceProvider>,
      {container},
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
