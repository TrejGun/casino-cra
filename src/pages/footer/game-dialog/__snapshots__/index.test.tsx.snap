// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameDialog /> renders dialog 1`] = `
<DocumentFragment>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root Panel-text-2 MuiTypography-body1"
                  >
                    ?
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root Panel-text-2 MuiTypography-body1"
                  >
                    ?
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root Panel-text-2 MuiTypography-body1"
                  >
                    ?
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              @
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="GameDialog-grow-1"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Close
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary Mui-focusVisible Mui-focusVisible"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Spin
            </span>
            <span
              class="MuiTouchRipple-root"
            >
              <span
                class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
              >
                <span
                  class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
                />
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
